connectors/sources/redis.py

```diff
@@ -148,11 +153,73 @@ async def ping(self):
         await self._client.ping()
 
 
+class RedisAdvancedRulesValidator(AdvancedRulesValidator):
+    RULES_OBJECT_SCHEMA_DEFINITION = {
+        "type": "object",
+        "properties": {
+            "database": {"type": "integer", "minLength": 1, "minimum": 0},
+            "key_pattern": {"type": "string", "minLength": 1},
+            "_type": {
+                "type": "string",
+                "minLength": 1,
+            },
+        },
+        "required": ["database"],
+        "anyOf": [
+            {"required": ["key_pattern"]},
+            {"required": ["_type"]},
+        ],
+        "additionalProperties": False,
+    }
+
+    SCHEMA_DEFINITION = {"type": "array", "items": RULES_OBJECT_SCHEMA_DEFINITION}
+
+    SCHEMA = fastjsonschema.compile(definition=SCHEMA_DEFINITION)
+
+    def __init__(self, source):
+        self.source = source
+
+    async def validate(self, advanced_rules):
+        if len(advanced_rules) == 0:
+            return SyncRuleValidationResult.valid_result(
+                SyncRuleValidationResult.ADVANCED_RULES
+            )
+        return await self._remote_validation(advanced_rules)
+
+    async def _remote_validation(self, advanced_rules):
+        try:
+            RedisAdvancedRulesValidator.SCHEMA(advanced_rules)
+        except fastjsonschema.JsonSchemaValueException as e:
+            return SyncRuleValidationResult(
+                rule_id=SyncRuleValidationResult.ADVANCED_RULES,
+                is_valid=False,
+                validation_message=e.message,
+            )
+        invalid_db = []
+        database_to_filter = [rule["database"] for rule in advanced_rules]
+        for db in database_to_filter:
+            check_db = await self.source.client.validate_database(db=db)
+            if not check_db:
+                invalid_db.append(db)
+                continue
```

Comment:
`continue` is not necessary here

I think you should avoid calling remotely if `db` is not a positive number.

Suggesting extract this to a separate method `validate_databases` so that  `_remote_validation` of both `RedisDataSource` and `RedisAdvancedRulesValidator` can make use of it.
https://github.com/elastic/connectors/blob/c180948f7425aefd0b124d0f634697e0e06334e7/connectors/sources/redis.py#L273-L287

Reply:
@wangch079 In RedisAdvancedRulesValidator we don't need to validate manually that db is positive or integer, w are validing it while compiling fastjsonschema on line 177. So even after seprating  out the code here. we'll not be able to use it in `RedisAdvancedRulesValidator`. What's your opinion on this?